fastchat/model/model_adapter.py

```diff
@@ -373,14 +373,25 @@ class PeftModelAdapter:
 
     def match(self, model_path: str):
         """Accepts any model path with "peft" in the name"""
-        return "peft" in model_path
+        if os.path.exists(os.path.join(model_path, "adapter_config.json")):
+            return True
+        else:
+            from huggingface_hub import HfApi
+
+            api = HfApi()
+            model_info = api.model_info(model_path)
+            repo_files = [file.rfilename for file in model_info.siblings]
+            if "adapter_config.json" in repo_files:
+                return True
```

Comment:
I wanted to share that I learned very recently that you can use an utility method from `huggingface_hub` called `list_repo_files`
```python
from huggingface_hub import list_repo_files

list_remote_files = list_repo_files(model_path, revision=revision, use_auth_token=use_auth_token)
```
This method can be very useful in case you want to load the model from a private repository (with the argument `use_auth_token`) or a model that lives on a separate branch (through the argument `revision`). 
Thanks !

Reply:
Thank you for providing this! Will try to utilize this :)